client/galaxy/scripts/components/Workflow/Run/WorkflowRun.vue

```diff
@@ -83,20 +110,49 @@ export default {
             runButtonWaitText: "",
             runButtonPercentage: -1,
             invocations: null,
+            simpleForm: null,
             model: null,
         };
     },
     created() {
         getRunData(this.workflowId)
             .then((runData) => {
                 const model = new WorkflowRunModel(runData);
+                let simpleForm = this.preferSimpleForm;
+                if (simpleForm) {
+                    // These only work with PJA - the API doesn't evaluate them at
+                    // all outside that context currently. The main workflow form renders
+                    // these dynamically and takes care of all the validation and setup details
+                    // on the frontend. If these are implemented on the backend at some
+                    // point this restriction can be lifted.
+                    if (model.hasReplacementParametersInToolForm) {
+                        console.log("cannot render simple workflow form - has ${} values in tool steps");
+                        simpleForm = false;
+                    }
+                    // If there are required parameters in a tool form (a disconnected runtime
+                    // input), we have to render the tool form steps and cannot use the
+                    // simplified tool form.
+                    if (model.hasOpenToolSteps) {
+                        console.log(
+                            "cannot render simple workflow form - one or more tools have disconnected runtime inputs"
```

Comment:
These probably should be shown as error messages on the screen.

Reply:
These aren't errors though (?) - if any of these conditions are met the simple the full form is just rendered. I was thinking about adding a "lint for best practices" button in the workflow editor - that might be a better spot for user messages around these?

Although - I am noticing I don't have that simple_form=false here in this branch like the others.

Reply:
Agreed it doesn't make sense to bubble this to the user; there's nothing for them to act on here and it's not something we can/should pause any interaction for.